### PR TITLE
Image input

### PR DESCRIPTION
**Pull request by mimansha-swarup**

### CODE REVIEW

**Review by srishti-maurya — APPROVED** [review]
UI looks amazing

File: components/image/image.css [hunk 1]
```diff
@@ -0,0 +1,17 @@
+.img-responsive{
+    width:100%;
+    height: auto;
+    object-fit: cover;
+}
+.img-rnd{
```
Comment (jainPrajwal) [inline components/image/image.css]: What is rnd?

File: components/image/image.css [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    border-radius: 50%;
+}
+.img-abs-rnd{
+    aspect-ratio: 1 /1;;
+    object-fit: cover;
+
+}
+/* Not-imp */
```
Comment (jainPrajwal) [inline components/image/image.css]: Remove unnecessary comments

File: components/input/input.css [hunk 3]
```diff
@@ -0,0 +1,52 @@
+.form-grp{
```
Comment (jainPrajwal) [inline components/input/input.css]: What is grp?

File: components/image/image.css [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+.img-cont{
```
Comment (jainPrajwal) [inline components/image/image.css]: What is cont?

File: components/input/input.css [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+.inpt-grp input:focus{
+    border: .1rem solid var(--Teal-0);
+    outline: none;
+}
+.inpt-grp input::placeholder{
+    color:var(--Teal-4);
+    
+}
+.success-inpt{
```
Comment (jainPrajwal) [inline components/input/input.css]: What is inpt?

File: components/input/input.css [hunk 6]
```diff
@@ -0,0 +1,52 @@
+.form-grp{
+    gap: 1rem
+}
+.inpt-grp{
+    position: relative;
+    gap: .25rem;
```
Comment (jainPrajwal) [inline components/input/input.css:13]: gap only works in flex and grid.

File: components/input/input.css [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    color: var(--Light-2);
```
Comment (jainPrajwal) [inline components/input/input.css]: Try giving more meaningful variable names.
For eg. --primary-light, --secondary-light

File: components/input/input.css [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+.inpt-grp input{
+    border: .1rem solid var(--Teal-2);
+    border-radius: .25rem;
+    font-size: 1rem;
+    padding: .25rem .75rem;
+    background-color: var(--Elevation--01);
```
Comment (jainPrajwal) [inline components/input/input.css]: Have consistent naming.
For eg. Teal-2 then it should be Elevation-1.

File: components/input/input.html [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Sinister-UI | Input</title>
+    <link rel="icon" type="image/x-icon" href="../../assets/Sinister-UI.svg">
+    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+    <link rel="stylesheet" href="../../components/main.css">
+    <!-- <link rel="stylesheet" href="../../components/typography/typography.css"> -->
```
Comment (jainPrajwal) [inline components/input/input.html]: Remove unnecessary comments

File: components/input/input.html [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+    <link rel="stylesheet" href="../../docs.css">
+    <link rel="stylesheet" href="../../landing.css">
+</head>
+<body>
+    <header class="navbar" >
+        <p class="navbar-brand">
+            <a href="../../index.html"  >Sinister UI</a>
+            <small class="subtitle2 small-text" >V1.0</small>
```
Comment (jainPrajwal) [inline components/input/input.html:18]: what does subtitle2 mean?
I suppose it should be subtitle-lg for large subtitle

File: components/input/input.html [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+                    <li>
+                        Installation
+                    </li>
+                </a>
+                <a href="#" class="btn btn-text btn-link teal button-txt" >
+                    <li>
+                        Color
+                    </li>
+                </a>
+                <a href="#" class="btn btn-text btn-link teal button-txt" >
```
Comment (jainPrajwal) [inline components/input/input.html]: You have used both `btn-text` and  `button-text`.
It seems similar.
Use proper naming conventions.

File: components/input/input.html [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+                    <li>
+                        Input
+                    </li>
+                </a>
+            </ul>
+        </aside>
+        
+        <div class="docs-body">
+            <h1 class="headline1 text-center">Input</h1>
+            <p class="body1">Input fields allow users to enter text into a UI. They typically appear in forms and dialogs.</p>
```
Comment (jainPrajwal) [inline components/input/input.html:88]: What is body1 and heading1?

File: components/input/input.html [hunk 13]
```diff
@@ -0,0 +1,196 @@
...
+                      </iframe>
+
+                    </code>
+                </div>
+            
+            </section>     
+            <hr class="line-horz">               
+            <section>
+
+                <h2 class="headline2">Sucess Input Field</h2>
```
Comment (jainPrajwal) [inline components/input/input.html]: Spell check Success

**Review by jainPrajwal — CHANGES_REQUESTED** [review]
Added suggestions and sanity checks

File: components/input/input.html (hunk 12 above)
Comment (mimansha-swarup) [reply components/input/input.html:88]: I am keeping typo in the form of
healdine1.....headline6, body1,body2, subtitle1,subtitle2
just like materialUI

File: components/image/image.css (hunk 1 above)
Comment (mimansha-swarup) [reply components/image/image.css]: changed to round

File: components/input/input.css (hunk 5 above)
Comment (mimansha-swarup) [reply components/input/input.css]: change to input

File: components/input/input.css (hunk 6 above)
Comment (mimansha-swarup) [reply components/input/input.css:13]: i was using flex class to add flex property and changing gap from this class
but changed this

File: components/image/image.html [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+            <hr class="line-horz">               
+            <section>
+
+                <h2 class="headline2">Round image</h2>
+                <p class="body1"> 
+                   to make image round or ellipse use 
+                    <span class="highlight-txt"> img-rnd </span>.
+                    if Image is not sqare and still  need round image use 
+                    <span class="highlight-txt"> img-rnd </span> and
+                    <span class="highlight-txt"> img-abs-rnd </span> together.
```
Comment (SAYUK09) [inline components/image/image.html]: Didn't understand what `abs` stands for in `img-abs-rnd`

File: components/input/input.css [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+    color: var(--Red-3);    
+}
+.error-inpt input{
+    border: .1rem solid var(--Red-2);    
+}
+.error-inpt input::placeholder{
+    color: var(--Red-4);    
+}
+.error-inpt input:focus{
+    border: .1rem solid var(--Red-0);
```
Comment (SAYUK09) [inline components/input/input.css]: I see `0.1` is used quite some time.
You can use variables for commonly used values like 0.1,2,4rem etc.

File: components/input/input.html (hunk 12 above)
Comment (SAYUK09) [reply components/input/input.html:88]: That works too. You may shift to something like **xl**, **lg**, **md**, **sm**, **xm**
It's your preference here.  But the above conventions look more appealing.

File: index.html [hunk 16]
```diff
@@ -32,9 +32,13 @@
             <img src="./assets/Sinister-UI.svg" alt="">
```
Comment (SAYUK09) [inline index.html]: Alway give a `alt` text for accessibility purposes

File: components/image/image.html [hunk 17]
```diff
@@ -0,0 +1,152 @@
...
+            <section>
+
+                <h2 class="headline2">Responsive image</h2>
+                <p class="body1"> 
+                   to meake image responeive use 
+                    <span class="highlight-txt"> img-responsive </span>.
+                </p>
+                <div class="showcase">
+                    <div class="img-cont">
+                        <img class="img-responsive" src="https://picsum.photos/400/400">
```
Comment (SAYUK09) [inline components/image/image.html:99]: use `alt` for all images.

**Review by SAYUK09 — CHANGES_REQUESTED** [review]
The code looks well structured. Suggested a few changes.
Try to keep your PR small. I see the PR has 12 files but as the name of the branch suggests it should have only files related to the **image** component.

Everything else is good.

**Pull request merged**